test/python/circuit/library/test_permutation.py

```diff
@@ -38,6 +40,19 @@ def test_permutation_bad(self):
         """Test that [0,..,n-1] permutation is required (no -1 for last element)."""
         self.assertRaises(CircuitError, Permutation, 4, [1, 0, -1, 2])
 
+    def test_get_ordered_swap(self):
+        """Test get_ordered_swap function."""
+        np.random.seed(1)
+        for length in range(5, 20, 2):
+            for _ in range(5):
+                permutation = np.random.permutation(length)
+                swap_list = _get_ordered_swap(permutation)
+                output = list(range(length))
+                for i, j in swap_list:
+                    output[i], output[j] = output[j], output[i]
+                self.assertTrue(np.array_equal(permutation, output))
+                self.assertLess(len(swap_list), length)
```

Comment:
is it possible to write a functional test here that checks the Operator implemented by this circuit is equal to the expected Operator of a Permutation? The unitary associated with a permutation can be obtained by permuting columns of a 2^n x 2^n Identity matrix (though I haven't worked out how exactly). This particular test seems to rely on `_get_ordered_swap` being correct, which seems like an implementation detail.

Reply:
ok